src/client/Client.cc

```diff
@@ -5628,7 +5628,7 @@ int Client::mount(const std::string &mount_root, const UserPerm& perms,
   }
 
   filepath fp(CEPH_INO_ROOT);
-  if (!mount_root.empty()) {
+  if (cct->_conf->client_mountpoint != std::string("/") && !mount_root.empty()) {
```

Comment:
This condition is the wrong way around, right?

If client_mountpoint is "/" (i.e. not over-ridden) then we *do* want to set metadata["root"] to the mount_root.  Whereas if client_mountpoint is *not* "/" (i.e. the user set it in ceph.conf or in ceph-fuse -r) then we do not want to set metadata["root"] again here.